etl/steps/data/garden/cedlas/2024-03-08/sedlac.py

```diff
@@ -0,0 +1,176 @@
+    tb_no_spells = tb_no_spells.set_index(["country", "year", "table"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir,
+        tables=[tb, tb_no_spells],
+        check_variables_metadata=True,
+        default_metadata=ds_meadow.metadata,
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def load_tables(dataset: Dataset, tables_dict: Dict) -> List[Table]:
+    """Load tables from the meadow dataset."""
+
+    tables = []
+    for t in tables_dict:
+        tb = dataset[t].reset_index()
+
+        # Create a column with the name of the table
+        tb["table"] = tables_dict[t]["column_name"]
+
+        tables.append(tb)
+
+    return tables
+
+
+def merge_tables(tables: List[Table], merge_or_concat: str, short_name: str) -> Table:
+    """Merge tables into a single table."""
+
+    if merge_or_concat == "concat":
+        # Concatenate tables
+        tb = pr.concat(tables, ignore_index=True, short_name=short_name)
+    elif merge_or_concat == "merge":
+        # use the first table as base
+        tb = tables[0]
+        for i in range(len(tables) - 1):
+            if i < len(tables) - 1:
+                # Merge tables
+                tb = pr.merge(
+                    tb,
+                    tables[i + 1],
+                    on=["country", "year", "survey_number", "survey", "table"],
+                    how="outer",
+                    short_name=short_name,
+                )
+
+    else:
+        raise ValueError("Invalid value for merge_or_concat: must be 'merge' or 'concat'.")
+
+    return tb
+
+
+def remove_spells_duplicates_and_add_urban(tb: Table, short_name: str) -> Table:
+    """Remove duplicate rows and spells from a table."""
+
+    tb = tb.copy()
+
+    # Make years int by selecting the text of the first 4 characters
+    tb["year"] = tb["year"].str[:4].astype(int)
+
+    # Remove duplicates
+    tb = tb.drop_duplicates(subset=["country", "year", "table"], keep="last", ignore_index=True)
+
+    # Add "(urban)" to country names when needed
+    # Make country string
+    tb["country"] = tb["country"].astype(str)
+
+    # For Argentina
+    mask = tb["country"] == "Argentina"
+    tb.loc[mask, "country"] = tb.loc[mask, "country"] + " (urban)"
+    # For Bolivia
+    mask = (tb["country"] == "Bolivia") & (tb["survey_number"] == 1) & (tb["year"] <= 1997)
+    tb.loc[mask, "country"] = tb.loc[mask, "country"] + " (urban)"
+    # For Uruguay
+    mask = (tb["country"] == "Uruguay") & (tb["survey_number"] == 1)
+    tb.loc[mask, "country"] = tb.loc[mask, "country"] + " (urban)"
+
+    # Drop survey columns
+    tb = tb.drop(columns=["survey", "survey_number"])
+
+    # Assign short_name to the table
+    tb.m.short_name = short_name
+
+    return tb
+
+
+def drop_urban_rural_from_poverty_table(tb: Table) -> Table:
```

Comment:
super optional and I can imagine you probably don't need these indicators, but was wondering if there might be some interesting things in these indicators - ie differences in inequalities in rural vs urban areas.